test/react-decor/react-decor-reflection.spec.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from "react";
+import { expect, sinon } from "test-drive-react";
+import { testWithBothComponentTypes } from "../test-drivers/test-tools";
+import { ElementArgs } from "../../src/react-decor/common";
+import { runInContext } from "../../src/core/config";
+import { decorateReactComponent, decorationReflection, StatefulElementHook, StatelessElementHook } from "../../src";
+
+type PropsWithName = { name: string };
+
+const SFComp: React.SFC<PropsWithName> = ({name}) => (
```

Comment:
over-complicated component for this test.
Either narrow down the render fucntion to the bare minimum, or extract it to a reusable test driver